ci_framework/roles/os_net_setup/tasks/main.yml

```diff
@@ -0,0 +1,40 @@
+- name: get openstack admin credentials from k8s
+  block:
+  - name: Get clouds.yaml configuration
+    kubernetes.core.k8s_info:
+      kubeconfig: "{{ k8s_kubeconfig }}"
+      context: "{{ k8s_context }}"
+      kind: ConfigMap
+      namespace: openstack
+      name: openstack-config
+    register: clouds_yaml
+
+  - name: Get the OSP admin password
+    kubernetes.core.k8s_info:
+      kubeconfig: "{{ k8s_kubeconfig }}"
+      context: "{{ k8s_context }}"
+      kind: Secret
+      namespace: openstack
+      name: openstack-config-secret
+    register: osp_secret
+  vars:
+      k8s_kubeconfig: "{{ cifmw_os_net_setup_kudeconfig_location | default(cifmw_install_yamls_vars.KUBECONFIG) | default(ansible_user_dir + '/.crc/machines/crc/kubeconfig') }}"
+      k8s_context: "{{ cifmw_os_net_setup_kudeconfig_context | default('admin') }}"
+
+- name: Construct auth param for openstack
+  ansible.builtin.set_fact:
+    openstack_auth: |
+      {{ ((clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default'] |
+      combine ({'auth': { 'password' : (osp_secret.resources[0].data['secure.yaml']|b64decode|from_yaml)['clouds']['default']['auth']['password']} }, recursive=true))['auth'] }}
+    region_name: "{{ (clouds_yaml.resources[0].data['clouds.yaml']| from_yaml)['clouds']['default']['region_name'] }}"
+
+- openstack.cloud.networks_info:
+    auth: "{{ openstack_auth }}"
+  register: net_info
+
+- name: Process network list element
+  ansible.builtin.include_tasks:
+    file: subtask_net.yml
+  loop: "{{ cifmw_os_net_setup_config|flatten(levels=1) }}"
```

Comment:
Any way to validate a bit that `cifmw_os_net_setup_config` parameter?

Reply:
Maybe json-schema somehow can work..

Reply:
This may be a follow-up.